File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    ans = [arr[0]] # 첫 인덱스는 그냥 넣어둠
    idx = 1 # 기준인덱스 지정
    
    for i in range(idx, len(arr)):
        if arr[i] != arr[i-1]:
            ans.append(arr[i])
            idx = i + 1
        
    return ans

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] temp = new int[arr.length];
        int idx = 0;
        
        temp[idx++] = arr[0];
        
        for(int i=1; i< arr.length; i++){
            if (arr[i] != arr[i-1]){
                temp[idx++] = arr[i];
            }
        }
        
        int[] result = new int[idx];
        for (int i = 0; i < idx; i++){
            result[i] = temp[i];
        }
        
        return result;
    }
}